omni-relayer/src/workers/solana.rs

```diff
@@ -0,0 +1,326 @@
+use std::{str::FromStr, sync::Arc};
+
+use anyhow::Result;
+use futures::future::join_all;
+use log::{error, info, warn};
+
+use omni_connector::OmniConnector;
+use omni_types::{
+    locker_args::StorageDepositAction, prover_args::WormholeVerifyProofArgs,
+    prover_result::ProofKind, ChainKind, OmniAddress,
+};
+use solana_sdk::pubkey::Pubkey;
+
+use crate::{config, utils};
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct InitTransferWithTimestamp {
+    pub amount: u128,
+    pub token: String,
+    pub recipient: String,
+    pub fee: u128,
+    pub native_fee: u64,
+    pub emitter: String,
```

Comment:
Emitter is represented as a regular Pubkey. In order to parse it as OmniAddress, we need to use OmniAddress::new_from_slice(ChainKind::Sol, &emitter.to_bytes()), so it can't be done automatically

Reply:
Do we need to parse it from JSON? Actually, we can. We can set up a custom parser for this specific field.

Google something like: serde deserialize_with